x/manifest/keeper/keeper.go

```diff
@@ -87,110 +87,37 @@ func (k *Keeper) ExportGenesis(ctx context.Context) *types.GenesisState {
 	}
 }
 
-func (k *Keeper) GetShareHolders(ctx context.Context) ([]*types.StakeHolders, error) {
-	params, err := k.Params.Get(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	return params.StakeHolders, nil
-}
-
-// IsManualMintingEnabled returns nil if inflation mint is 0% (disabled)
-func (k Keeper) IsManualMintingEnabled(ctx context.Context) bool {
-	params, err := k.Params.Get(ctx)
-	if err != nil {
-		panic(err)
-	}
-
-	return !params.Inflation.AutomaticEnabled
-}
-
-type StakeHolderPayout struct {
-	Address string
-	Coin    sdk.Coin
-}
-
-// Returns the amount of coins to be distributed to the holders
-func (k Keeper) CalculateShareHolderTokenPayout(ctx context.Context, c sdk.Coin) ([]StakeHolderPayout, error) {
-	sh, err := k.GetShareHolders(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	pairs := make([]StakeHolderPayout, 0, len(sh))
-
-	for _, s := range sh {
-		s := s
-		pct := sdkmath.NewInt(int64(s.Percentage)).ToLegacyDec().QuoInt64(types.MaxPercentShare)
-		coinAmt := pct.MulInt(c.Amount).RoundInt()
-
-		if coinAmt.IsZero() {
-			// too small of an amount to matter (< 1 utoken)
-			continue
-		}
-
-		pairs = append(pairs, StakeHolderPayout{
-			Address: s.Address,
-			Coin:    sdk.NewCoin(c.Denom, coinAmt),
-		})
-
-	}
-
-	return pairs, nil
-}
-
-// PayoutStakeholders mints coins and sends them to the stakeholders.
-// This is called from the endblocker, so panics should never happen.
-// If it does, something is very wrong w/ the SDK. Any logic specific to auto minting
-// should be kept out of this to properly handle and return nil instead.
-func (k Keeper) PayoutStakeholders(ctx context.Context, c sdk.Coin) error {
-	pairs, err := k.CalculateShareHolderTokenPayout(ctx, c)
-	if err != nil {
-		return err
-	}
-
-	if err := k.bankKeeper.MintCoins(ctx, types.ModuleName, sdk.NewCoins(c)); err != nil {
-		return err
-	}
+// PayoutStakeholders mints and sends coins to stakeholders.
+func (k Keeper) PayoutStakeholders(ctx context.Context, payouts []types.PayoutPair) error {
```

Comment:
nit: Rename to `Payout`

Reply:
https://github.com/liftedinit/manifest-ledger/pull/68/commits/2615b2d4a387a22ebea80d5eee2e43d64015b410